### ISSUE TITLE
Warnings for several karma packages

### ISSUE THREAD

**Issue by timkindberg** [ngbp/ngbp#58]
I'm getting the following error I think during `bower install`, it may have been `npm install` though. Sorry my cmd won't scroll any higher to let me see :P

```
Running "karma:continuous" (karma) task
WARN [plugin]: Cannot find plugin "karma-jasmine".
  Did you forget to install it ?
  npm install karma-jasmine --save-dev
WARN [plugin]: Cannot find plugin "karma-firefox-launcher".
  Did you forget to install it ?
  npm install karma-firefox-launcher --save-dev
WARN [plugin]: Cannot find plugin "karma-chrome-launcher".
  Did you forget to install it ?
  npm install karma-chrome-launcher --save-dev
WARN [plugin]: Cannot find plugin "karma-coffee-preprocessor".
  Did you forget to install it ?
  npm install karma-coffee-preprocessor --save-dev
Warning: No provider for "framework:jasmine"! (Resolving: framework:jasmine) Use
 --force to continue.

Aborted due to warnings.
```

I think you are trying to load these in a clever way at the bottom of the Gruntfile.js? Perhaps its because I use Windows 7. Which I know you'll probably scoff at and that's fine. I had to install them manually with:

```
npm install karma-jasmine karma-firefox-launcher karma-chrome-launcher karma-
coffee-preprocessor --save-dev
```

Might be good for a FAQ at least or something...

**Comment by joshdmiller** [comment]
Those errors are thrown during the build itself - specifically during 'grunt build'. Karma has several peer dependencies to make it work, which it is listing in the error. The problem is most likely not on the ngBoilerplate side. Are you running at least Node.js 0.8.19? And you ran `npm install` before running `grunt build`, right?

**Comment by ElliotPsyIT** [comment]
I just received the identical karma error after installing ng-boilerplate and completing the install instructions, all of which went fine including `npm install`.  I then ran `grunt watch` and got the error.  All the other grunt tasks under grunt watch seemed to run fine up until the error.  My dev environment is Mac OS X 10.8.4, nodejs v0.8.9, Karma version: 0.8.6, grunt-cli v0.1.6, grunt v0.4.1, bower 0.9.2.  My apologies, I am an angular and a grunt nOOB so I don't know enough to do much further troubleshooting independently.

**Comment by timkindberg** [comment]
Sorry meant to reply sooner. I did have the latest npm (1.1.49). I did run npm install first. I was following your instructions on the main readme.md step by step. I couldn't remember when I got the errors so I think they probably happened during `grunt watch` for me as well and I just thought I remembered them happening during `bower install`.

**Comment by ElliotPsyIT** [comment]
Well, maybe I can do a little troubleshooting independently :)

In looking at the karma package.json on github, the plugins referenced in the error are listed in the 'peerDependencies' section. 

```
"peerDependencies": {
    "karma-jasmine": "*",
    "karma-requirejs": "*",
    "karma-coffee-preprocessor": "*",
    "karma-chrome-launcher": "*",
    "karma-firefox-launcher": "*",
    "karma-phantomjs-launcher": "*",
    "karma-script-launcher": "*"
  },
```

But, those peer dependencies were not loaded, apparently, when I installed karma 0.8.6.  When I look at karma's node_modules directory on my dev machine (/usr/local/lib/node_modules/karma), the package_json does not have a 'peerDependencies' section.  As a nodejs/npm nOOb too :), this may be correct for all I know, but I don't see how the peer dependencies come along when you install karma if they aren't listed in the package.json?  I must be missing something simple.  Any help to a newcomer would be appreciated!

**Comment by joshdmiller** [comment]
@fielstem - The problem is that your node is too old. Peer dependencies were introduce starting in version 0.8.19 and you're running 0.8.9. You can read more about them [here](http://blog.nodejs.org/2013/02/07/peer-dependencies/).

Unfortunately the new version of karma - as well as other packages - make use of this cool new feature. I'll bet @timkindberg's problem is the node version as well.

I should surely add a note about the node version requirement to the readme as some people aren't yet running the new version. But it's not that new - it was release back in January or February.

**Comment by ElliotPsyIT** [comment]
Thanks very much for the information.  I just upgraded node to v0.10.12 and installed karma.  The install completed.  But in various places I noticed warnings about there being no repository field.  Here's a list of them that occurs right before the karma install completes.  And, there is no change in the error when running `grunt watch`, probably due to this issue limiting a incomplete karma install.

```
npm WARN package.json faye-websocket@0.4.3 No repository field.
npm WARN package.json faye-websocket@0.4.3 'repositories' (plural) Not supported.
npm WARN package.json Please pick one as the 'repository' field
npm WARN package.json bower@0.3.2 No repository field.
npm WARN package.json requirejs@2.0.6 No repository field.
npm WARN package.json cli-table@0.2.0 No repository field.
npm WARN package.json cheerio-select@0.0.3 No repository field.
npm WARN package.json stable@0.1.2 No repository field.
npm WARN package.json vows@0.6.4 No repository field.
npm WARN package.json debug@0.7.0 No repository field.
npm WARN package.json formidable@1.0.9 No repository field.
npm WARN package.json jshint@0.7.3 No repository field.
npm WARN package.json growl@1.7.0 No repository field.
npm WARN package.json github-url-from-git@1.1.1 No repository field.
npm WARN package.json eyes@0.1.8 No repository field.
npm WARN package.json assert-plus@0.1.2 No repository field.
npm WARN package.json ctype@0.5.2 No repository field.
npm WARN package.json amdefine@0.0.4 No repository field.
```

(Sorry to burden you with helping debug my problems with node and karma rather than ng-boilerplate - clearly it's newbie issues, but I certainly appreciated your willingness to help).

**Comment by joshdmiller** [comment]
@fielstem No worries - I am not burdened. :-) 

The repository errors are ignorable. The new node likes packages to define it, but doesn't require it and so will warn when it's missing.

If you still get the same errors about the missing karma packages, can you post the output of `npm list`? It will be long.

**Comment by ElliotPsyIT** [comment]
This is very good of you!  Here's the output of `npm list`

```
$ npm list
npm WARN package.json dateformat@1.0.2-1.2.3 No repository field.
npm WARN package.json jshint@1.1.0 No repository field.
npm WARN package.json vows@0.7.0 No repository field.
npm WARN package.json peakle@0.0.1 No repository field.
npm WARN package.json faye-websocket@0.4.4 No repository field.
npm WARN package.json faye-websocket@0.4.4 'repositories' (plural) Not supported.
npm WARN package.json Please pick one as the 'repository' field
npm WARN package.json pause@0.0.1 No repository field.
npm WARN package.json eyes@0.1.8 No repository field.
npm WARN package.json policyfile@0.0.4 No repository field.
npm WARN package.json policyfile@0.0.4 'repositories' (plural) Not supported.
npm WARN package.json Please pick one as the 'repository' field
ng-boilerplate@0.3.0 /Users/Me/Angular/ng-boilerplate-test/ng-boilerplate
├─┬ grunt@0.4.1
│ ├── async@0.1.22
│ ├── coffee-script@1.3.3
│ ├── colors@0.6.0-1
│ ├── dateformat@1.0.2-1.2.3
│ ├── eventemitter2@0.4.12
│ ├─┬ findup-sync@0.1.2
│ │ └── lodash@1.0.1
│ ├─┬ glob@3.1.21
│ │ ├── graceful-fs@1.2.2
│ │ └── inherits@1.0.0
│ ├── hooker@0.2.3
│ ├── iconv-lite@0.2.10
│ ├─┬ js-yaml@2.0.5
│ │ ├─┬ argparse@0.1.15
│ │ │ ├── underscore@1.4.4
│ │ │ └── underscore.string@2.3.1
│ │ └── esprima@1.0.3
│ ├── lodash@0.9.2
│ ├─┬ minimatch@0.2.12
│ │ ├── lru-cache@2.3.0
│ │ └── sigmund@1.0.0
│ ├─┬ nopt@1.0.10
│ │ └── abbrev@1.0.4
│ ├─┬ rimraf@2.0.3
│ │ └── graceful-fs@1.1.14
│ ├── underscore.string@2.2.0rc
│ └── which@1.0.5
├─┬ grunt-bump@0.0.6
│ └── semver@1.1.4
├─┬ grunt-coffeelint@0.0.6
│ └─┬ coffeelint@0.5.6
│   ├── coffee-script@1.4.0
│   ├─┬ glob@3.1.14
│   │ ├── graceful-fs@1.1.14
│   │ ├── inherits@1.0.0
│   │ └─┬ minimatch@0.2.9
│   │   ├── lru-cache@2.0.4
│   │   └── sigmund@1.0.0
│   ├─┬ optimist@0.3.5
│   │ └── wordwrap@0.0.2
│   └─┬ vows@0.7.0
│     ├── diff@1.0.4
│     └── eyes@0.1.8
├── grunt-contrib-clean@0.4.1
├─┬ grunt-contrib-coffee@0.7.0
│ └── coffee-script@1.6.3
├── grunt-contrib-concat@0.3.0
├── grunt-contrib-copy@0.4.1
├─┬ grunt-contrib-jshint@0.4.3
│ └─┬ jshint@1.1.0
│   ├─┬ cli@0.4.4-2
│   │ └─┬ glob@3.2.1
│   │   ├── graceful-fs@1.2.2
│   │   └── inherits@1.0.0
│   ├── esprima@1.1.0-dev
│   ├─┬ minimatch@0.2.12
│   │ ├── lru-cache@2.3.0
│   │ └── sigmund@1.0.0
│   ├── peakle@0.0.1
│   ├── shelljs@0.1.4
│   └── underscore@1.4.4
├─┬ grunt-contrib-uglify@0.2.2
│ ├─┬ grunt-lib-contrib@0.6.1
│ │ └── zlib-browserify@0.0.1
│ └─┬ uglify-js@2.3.6
│   ├── async@0.2.9
│   ├─┬ optimist@0.3.7
│   │ └── wordwrap@0.0.2
│   └─┬ source-map@0.1.25
│     └── amdefine@0.0.5
├─┬ grunt-contrib-watch@0.4.4
│ ├─┬ gaze@0.3.4
│ │ ├─┬ fileset@0.1.5
│ │ │ └─┬ glob@3.2.1
│ │ │   ├── graceful-fs@1.2.2
│ │ │   └── inherits@1.0.0
│ │ └─┬ minimatch@0.2.12
│ │   ├── lru-cache@2.3.0
│ │   └── sigmund@1.0.0
│ └─┬ tiny-lr@0.0.4
│   ├── debug@0.7.2
│   ├── faye-websocket@0.4.4
│   ├─┬ noptify@0.0.3
│   │ └─┬ nopt@2.0.0
│   │   └── abbrev@1.0.4
│   └── qs@0.5.6
├─┬ grunt-conventional-changelog@0.1.2
│ └── shelljs@0.1.4
├── grunt-html2js@0.1.4
├─┬ grunt-karma@0.5.0
│ └─┬ karma@0.9.4
│   ├── chokidar@0.6.2
│   ├── coffee-script@1.6.3
│   ├── colors@0.6.0-1
│   ├── di@0.0.1
│   ├─┬ glob@3.1.21
│   │ └── inherits@1.0.0
│   ├── graceful-fs@1.2.2
│   ├─┬ http-proxy@0.10.3
│   │ ├── pkginfo@0.2.3
│   │ └─┬ utile@0.1.7
│   │   ├── async@0.1.22
│   │   ├── deep-equal@0.0.0
│   │   ├── i@0.3.1
│   │   ├── mkdirp@0.3.5
│   │   ├── ncp@0.2.7
│   │   └── rimraf@1.0.9
│   ├── lodash@1.1.1
│   ├─┬ log4js@0.6.6
│   │ ├── async@0.1.15
│   │ ├── dequeue@1.0.3
│   │ ├── readable-stream@1.0.2
│   │ └── semver@1.1.4
│   ├── mime@1.2.9
│   ├─┬ minimatch@0.2.12
│   │ ├── lru-cache@2.3.0
│   │ └── sigmund@1.0.0
│   ├─┬ optimist@0.3.7
│   │ └── wordwrap@0.0.2
│   ├── pause@0.0.1
│   ├── q@0.9.6
│   ├── rimraf@2.1.4
│   ├─┬ socket.io@0.9.16
│   │ ├── base64id@0.1.0
│   │ ├── policyfile@0.0.4
│   │ ├── redis@0.7.3
│   │ └─┬ socket.io-client@0.9.16
│   │   ├─┬ active-x-obfuscator@0.0.1
│   │   │ └── zeparser@0.0.5
│   │   ├── uglify-js@1.2.5
│   │   ├─┬ ws@0.4.27
│   │   │ ├── commander@0.6.1
│   │   │ ├── options@0.0.5
│   │   │ └── tinycolor@0.0.1
│   │   └── xmlhttprequest@1.4.2
│   └─┬ useragent@2.0.6
│     └── lru-cache@2.2.4
├─┬ grunt-ngmin@0.0.2
│ └─┬ ngmin@0.3.8
│   ├── clone@0.1.9
│   ├─┬ commander@1.1.1
│   │ └── keypress@0.1.0
│   ├─┬ escodegen@0.0.23
│   │ ├── estraverse@0.0.4
│   │ └─┬ source-map@0.1.25
│   │   └── amdefine@0.0.5
│   └── esprima@1.0.3
└─┬ grunt-recess@0.3.3
  ├─┬ grunt-lib-contrib@0.5.3
  │ └── zlib-browserify@0.0.1
  └─┬ recess@1.1.8
    ├── colors@0.6.0-1
    ├─┬ less@1.3.3
    │ └── ycssmin@1.0.1
    ├─┬ nopt@2.1.1
    │ └── abbrev@1.0.4
    ├── underscore@1.4.4
    └── watch@0.8.0
```

**Comment by ElliotPsyIT** [comment]
Ok.  I updated my version of node to the latest version which resolved the peerDependency problem and the installation completed successfully.  I did have one other peerDependency versioning hiccup during the installation.  This time it was between bower and a previously installed yeoman.  

```
error peerinvalid The package bower does not satisfy its siblings' peerDependencies requirements!
error peerinvalid Peer yo@1.0.0-rc.1 wants bower@~0.9.2
```

Checking the yo git repository, I found an update of a couple of days ago to yo@1.0.0-rc.1.1, with the sole change to the peerDependencies for bower from 0.9.2 to 0.9.3.  I updated yo and then finished installing ng-boilerplate without further ado.  

For this newbie, it's a wake up call to attend more closely to the version of the tools built on node that I'm using and the version of their interdependencies.  Thanks Josh for this great angular tool and for your generous assistance!!
